File: Android/src/com/bu6ido/jsantase/players/ConsolePlayer.java
/**
 * 
 */
package com.bu6ido.jsantase.players;

import java.util.Scanner;

import com.bu6ido.jsantase.Game;
import com.bu6ido.jsantase.common.PlayCard;

/**
 * @author bu6ido
 *
 */
public class ConsolePlayer extends AbstractPlayer 
{
	private Scanner scanner;
	public ConsolePlayer(Game game, String name) 
	{
		super(game, name);
		scanner = new Scanner(System.in);
	}

	@Override
	public void play() 
	{
		super.play();
		
		PlayCard card = null;
		int choice;
		String str;
		isChoosing = true;
		do
		{
			System.out.print("Please choose from (1.." + cards.size() + ", (C)lose, Change (T)rump): ");
			str = scanner.next();
			if (str.equalsIgnoreCase("c"))
			{
				game.closeGame(this);
				continue;
			}
			if (str.equalsIgnoreCase("t"))
			{
				game.changeTrump(this);
				game.getNotification().info(this + " changed the trump card.");
				continue;
			}
			try
			{
				choice = Integer.parseInt(str);
			}
			catch (NumberFormatException nfe)
			{
				choice = 0;
			}
			if ((choice >= 1) && (choice <= cards.size()))
			{
				card = cards.get(choice - 1);
			}
			else
			{
				card = null;
			}
		} 
		while (!game.validateCard(this, card, true));
		isChoosing = false;
		
		game.getNotification().info(this + " chose card " + card);
		
		game.nextMove(card);
	}

}

File: Qt/console/console_player.cpp
#include "console_player.h"

ConsolePlayer::ConsolePlayer(Game *game, string name) : AbstractPlayer(game, name)
{
//  cout << "Constructing ConsolePlayer" << endl;
}

ConsolePlayer::~ConsolePlayer()
{
//  cout << "Destroying ConsolePlayer" << endl;
}

void ConsolePlayer::play()
{
  AbstractPlayer::play();

  Card *card = NULL;
  int choice;
  string str;

  isChoosing = true;
  do
  {
    cout << "Please choose card (1.." << cards.size() << ", (C)lose, Change (T)rump): ";
    cin >> str;
    if (str.compare("c") == 0)
    {
      game->closeGame(this);
      continue;
    }
    if (str.compare("t") == 0)
    {
      game->changeTrump(this);
      continue;
    }
    choice = atoi(str.c_str());
    if ((choice >= 1) && (choice <= cards.size()))
    {
      card = cards.at(choice-1);
    }
    else
    {
      card = NULL;
    }
  }
  while (!game->validateCard(this, card, true));

  game->getBoard()->fireCardMove(card);

  isChoosing = false;

  stringstream ss;
  ss.str("");
  ss << *this << " chose card " << *card;
  game->getNotification()->info(ss.str());

  game->nextMove(card);
}

